include/drivers/i2s.h

```diff
@@ -520,6 +532,16 @@ static inline int z_impl_i2s_trigger(const struct device *dev,
 	const struct i2s_driver_api *api =
 		(const struct i2s_driver_api *)dev->api;
 
+	if (!IS_ENABLED(CONFIG_I2S_DIR_BOTH_SUPPORTED) && dir == I2S_DIR_BOTH) {
+		int ret = api->trigger(dev, I2S_DIR_RX, cmd);
+
+		if (ret < 0) {
+			return ret;
+		}
+
+		return api->trigger(dev, I2S_DIR_TX, cmd);
```

Comment:
This is going to be problematic. Assuming we are sending I2S_TRIGGER_START and the hardware is wired to start I2S clocks when RX starts by the time we trigger TX the physical transmission on the bus has started without correct TX data.

It may be better to simply add I2S_DIR_BOTH to the enum and update the documentation telling users that it is not supported by all the drivers. I.e. remove CONFIG_I2S_DIR_BOTH_SUPPORTED and the above code.

Reply:
Agreed. Removed `CONFIG_I2S_DIR_BOTH_SUPPORTED`.